State.cpp

```diff
@@ -0,0 +1,248 @@
+#include "BreathData.h"
+#include "State.h"
+#include "Constants.h"
+#include "Display.h"
+#include "Flow.h"
+#include "Pressure.h"
+#include "Valve.h"
+#include "ProportionalValve.h"
+#include "AlarmManager.h"
+#include "State.h"
+
+// Initialize singleton instances
+OffState* OffState::instance = 0;
+InsStateVC* InsStateVC::instance = 0;
+InsHoldStateVC* InsHoldStateVC::instance = 0;
+ExpStateVC* ExpStateVC::instance = 0;
+PeepStateVC* PeepStateVC::instance = 0;
+ExpHoldStateVC* ExpHoldStateVC::instance = 0;
+
+BreathData* State::breath;
+
+/*********************************************************/
+/* OffState                                              */
+/*********************************************************/
+
+// performs actions necessary upon entering a state
+static State* OffState::begin(BreathData b) {
+    if (instance==0) instance = new OffState();  // one-time setup
+    breath = &b;                                 // store reference to breath data
+
+    // valve settings
+    inspValve.endBreath(); // close the inspiratory valve
+    expValve.open();       // keep expiratory valve open for safety (also does not use as much power)
+
+    return instance;
+}
+
+static State* OffState::enter() {
```

Comment:
OK, at this point, I think you are mostly copying from the old state machine.  I have to go now, but this looks like it's on the right track.